### PR TITLE
fix writing of addresses.json

### PR DESCRIPTION
**Pull request by robert-e-davidson3**

### CODE REVIEW

File: packages/augur-tools/src/flash/scripts.ts [hunk 1]
```diff
@@ -40,11 +40,9 @@ export function addScripts(flash: FlashSession) {
       },
     ],
     async call(this: FlashSession, args: FlashArguments) {
-      const writeArtifacts = args.write_artifacts as boolean;
```
Comment (robert-e-davidson3) [inline packages/augur-tools/src/flash/scripts.ts:43]: random cleanup

File: packages/augur-tools/src/flash/flash.ts [hunk 2]
```diff
@@ -59,18 +59,24 @@ export class FlashSession {
...
+
       if (option.required) {
-        const arg = args[option.name];
         if (typeof arg === "undefined") {
           this.log(`ERROR: Must specify "--${option.name}"`);
           return;
         }
       }
+
+      const formattedName = option.name.replace('-', '_');
```
Comment (robert-e-davidson3) [inline packages/augur-tools/src/flash/flash.ts:75]: fixed discrepancy between interactive and non-interactive arguments containing dashes (`-`).

File: packages/augur-artifacts/src/index.ts [hunk 3]
```diff
@@ -55,7 +56,7 @@ export interface NetworkContractAddresses {
 }
 
 export async function setAddresses(networkId: NetworkId, addresses: ContractAddresses): Promise<void> {
-  const filepath = `${__dirname}/addresses.json`;
+  const filepath = path.join(__dirname, "../src/addresses.json"); // be sure to be in src dir, not build
```
Comment (robert-e-davidson3) [inline packages/augur-artifacts/src/index.ts:59]: should be overwriting source file addresses.json, not the one in the build dir... this was the bug

File: packages/augur-artifacts/src/index.ts [hunk 4]
```diff
@@ -64,5 +65,5 @@ export async function setAddresses(networkId: NetworkId, addresses: ContractAddr
 
   contents[networkId] = addresses;
 
-  await writeFile(filepath, JSON.stringify(contents, null, 2), "utf8");
+  await writeFile(filepath, JSON.stringify(contents, null, 1), "utf8");
```
Comment (robert-e-davidson3) [inline packages/augur-artifacts/src/index.ts:68]: for some reason addresses.json is formatted with one space of indentation :/

**Review by petong — APPROVED** [review]

**Pull request merged**